slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/JdbcMiscTest.scala

```diff
@@ -63,14 +63,34 @@ class JdbcMiscTest extends AsyncTest[JdbcTestDB] {
       GetStatementParameters.map { csp => csp shouldBe sp }
 
     Action.seq(
-      check(JdbcBackend.StatementParameters(ResultSetType.Auto, ResultSetConcurrency.Auto, ResultSetHoldability.Auto)),
+      check(JdbcBackend.StatementParameters(ResultSetType.Auto, ResultSetConcurrency.Auto, ResultSetHoldability.Auto, null)),
       Action.seq(
-        check(JdbcBackend.StatementParameters(ResultSetType.ScrollInsensitive, ResultSetConcurrency.Auto, ResultSetHoldability.Auto)),
-        check(JdbcBackend.StatementParameters(ResultSetType.ScrollInsensitive, ResultSetConcurrency.Auto, ResultSetHoldability.HoldCursorsOverCommit)).
+        check(JdbcBackend.StatementParameters(ResultSetType.ScrollInsensitive, ResultSetConcurrency.Auto, ResultSetHoldability.Auto, null)),
+        check(JdbcBackend.StatementParameters(ResultSetType.ScrollInsensitive, ResultSetConcurrency.Auto, ResultSetHoldability.HoldCursorsOverCommit, null)).
           withStatementParameters(rsHoldability = ResultSetHoldability.HoldCursorsOverCommit),
-        check(JdbcBackend.StatementParameters(ResultSetType.ScrollInsensitive, ResultSetConcurrency.Auto, ResultSetHoldability.Auto))
+        check(JdbcBackend.StatementParameters(ResultSetType.ScrollInsensitive, ResultSetConcurrency.Auto, ResultSetHoldability.Auto, null))
       ).withStatementParameters(rsType = ResultSetType.ScrollInsensitive),
-      check(JdbcBackend.StatementParameters(ResultSetType.Auto, ResultSetConcurrency.Auto, ResultSetHoldability.Auto))
+      check(JdbcBackend.StatementParameters(ResultSetType.Auto, ResultSetConcurrency.Auto, ResultSetHoldability.Auto, null))
+    )
+  }
+
+  def testOverrideStatements = {
+    class T(tag: Tag) extends Table[Int](tag, u"t") {
+      def id = column[Int]("a")
+      def * = id
+    }
+    val t = TableQuery[T]
+
+    val a1 = t.filter(_.id === 1)
+    val a2 = t.filter(_.id === 2)
+
+    seq(
+      t.schema.create,
+      t ++= Seq(1, 2, 3),
+      a1.result.map(_ shouldBe Seq(1)),
+      a1.result.overrideStatements(a2.result.statements).map(_ shouldBe Seq(2)),
```

Comment:
Great so see this coming. The most common use case of this is probably patching the SQL with operators not supported in slick directly, such as index hints. Or with dynamically generated filters. I slightly dislike that statements is a Sequence, but in this case it is known, that it has to have exactly one element and if it doesn't, the .head calls in the Slick code will throw exceptions, which can be hard to understand. Having a special Actions type for select queries could solve this, but would complicate the architecture. How about making overrideStatements take varargs, def overrideStatements(_statements: String*) ? Or maybe even def overrideStatements(_statement: String, _statements: String*) to force at least one.